### ISSUE TITLE
Pass the polygon data to `self._viewer.add_shapes()` in a more efficient way

### ISSUE THREAD

**Issue by LucaMarconato** [scverse/napari-spatialdata#56]
Right now we have the polygons data inside a geodataframe object, and we convert this to a list of list of floats to pass it to `self._viewer.add_shapes()` with this code
```
coordinates = polygons.geometry.apply(lambda x: np.array(x.exterior.coords).tolist()).tolist()
```
which takes 4 seconds for 160K polygons.

We have to check if there is a better way to pass this data to napari, maybe the ragged array representation.

**Comment by melonora** [comment]
Would you think of using async?

**Comment by LucaMarconato** [comment]
Yes, this could be an option. But in general even remaining single-threaded I think that there must be a more performant way to pass the data to napari without having to change data type and create a lot of new list objects.
